vision/emergent_object/emergent_object_processing.py

```diff
@@ -0,0 +1,161 @@
+                        subject_humanoid.attributes["longitude"],
+                    ),
+                ).feet
+
+                min_humanoid_distance = min(humanoid_distance, min_humanoid_distance)
+
+        # invert the humanoid distance to prioritize closest humanoids
+        min_humanoid_distance *= -1
+
+        evaluation: tuple[float, float, float, float] = (
+            min_odlc_distance,
+            min_humanoid_distance,
+            subject_humanoid.attributes["bounding_box_area"],
+            subject_humanoid.attributes["confidence"],
+        )
+
+        evaluations_list.append(evaluation)
+
+    evaluation_array: NDArray[Shape["*, 4"], Float64] = np.array(evaluations_list)
+
+    # Get the minimums of each category
+    evaluation_mins: NDArray[Shape["4"], Float64] = np.amin(evaluation_array, axis=0)
+
+    # Get the maximums of each category
+    evaluation_maxes: NDArray[Shape["4"], Float64] = np.amax(evaluation_array, axis=0)
+
+    # An error here could mean that all candidates were detected in the same image - This
+    #   is okay to ignore
+    with np.errstate(invalid="ignore"):
+        ranges: NDArray[Shape["4"], Float64] = evaluation_maxes - evaluation_mins
+
+    # Suppress warnings when getting 0/0 errors - This is okay because that means that
+    #   all candidates scored the same in that category
+    with np.errstate(invalid="ignore"):
+        normalized_evaluations: NDArray[Shape["*, 4"], Float64] = (
+            evaluation_array - evaluation_mins
+        ) / ranges
+
+    weighted_evaluations: NDArray[Shape["*, 4"], Float64] = (
+        normalized_evaluations * category_weights
+    )
+
+    # Use nansum to ignore categories with divide by zero errors
+    scores: NDArray[Shape["*"], Float64] = np.nansum(weighted_evaluations, axis=1)
+
+    return humanoids[np.argmax(scores)]
+
+
+if __name__ == "__main__":
+    saved_odlcs: ODLC_Dict = {
+        0: {"latitude": 52.10025, "longitude": 20.21222},
+        3: {"latitude": 52.80085, "longitude": 21.91021},
+    }
+
+    saved_humanoids: list[BoundingBox] = []
+
+    empty_vertices: Vertices = (
+        (0, 0),
+        (0, 0),
+        (0, 0),
+        (0, 0),
+    )
+
+    emergent_1: BoundingBox = BoundingBox(vertices=empty_vertices, obj_type=ObjectType.EMG_OBJECT)
+    emergent_1.attributes = {
+        "confidence": 0.5,
+        "bounding_box_area": 2,
+        "latitude": 52.2296756,
+        "longitude": 21.0122287,
+        "image_path": "image2.jpg",
+    }
+    saved_humanoids.append(emergent_1)
+
+    emergent_2: BoundingBox = BoundingBox(vertices=empty_vertices, obj_type=ObjectType.EMG_OBJECT)
+    emergent_2.attributes = {
+        "confidence": 0.3,
+        "bounding_box_area": 1,
+        "latitude": 52.406374,
+        "longitude": 16.9251681,
+        "image_path": "image1.jpg",
+    }
+    saved_humanoids.append(emergent_2)
+
+    pick_emergent_object(saved_humanoids, saved_odlcs)
```

Comment:
You aren't doing anything in main with the result of this function call, so it just gets immediately thrown out without the user knowing what it returned.